segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -1048,14 +1047,7 @@ private void truncateIndexSegment(String segment, long offset) {
             if (indexSegmentOffset == 0) {
                 return;
             }
-            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT)
-                    .whenComplete((v, exp) -> {
-                        if (exp != null) {
-                            log.warn("Failed to truncate index segment {} at offset {} due to", indexSegment, indexSegmentOffset, exp);
-                        } else {
-                            log.info("Successfully truncated index segment {} at offset {}", indexSegment, indexSegmentOffset);
-                        }
-                    });
+            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT);
```

Comment:
I think this method should return CompletableFuture from here.

Reply:
Method is returning  CompletableFuture<Void>, So I think it is fine here.

Reply:
**void truncateIndexSegment** its of void type.

Reply:
yes